lib/redix/utils.ex

```diff
@@ -112,6 +112,18 @@ defmodule Redix.Utils do
     end
   end
 
+  @doc """
+  This function unwraps the actual reason if an 'unknown POSIX error' is returned
+  from :inet.format_error/1
+  """
+  @spec format_error(term) :: String.t
+  def format_error(reason) do
+    case :inet.format_error(reason) do
+      'unknown POSIX error' -> inspect(reason)
+      message -> message
```

Comment:
Should I convert this to `binary`? Or update the type spec to be `String.t | charlist`?